### ISSUE TITLE
result reducers ignore __typename

### ISSUE THREAD

**Issue by Torsten85** [apollostack/apollo-client#870]
Hey,

using the result reducer, the `previousResult` objects has no `__typename` properties which results in a not working react component for me.

Manually adding the `__typename` to the query works though. Seems like the `addTypenameToDocument` functions is not called in this case.

**Comment by helfer** [comment]
@Torsten85 Can you clarify what you're expecting to see and what you're seeing instead? Is there an error message? I think I know what you mean, but it would be good if I could get more specifics before I look into it.

**Comment by Torsten85** [comment]
ok, lets say we have this mutation:

**Comment by Torsten85** [comment]
ok lets say we have this query & mutation:
```
query MyQuery {
  users {
    id,
   name
}

mutation MyMutation ($name: String!) {
  createUser(name: $name) {
    id,
   name
  }
}
```

if we use this mutation in with a graphql `updateQueries` function, we get 
```
[
  {
   id: "someid"
   name: "somename"
   __typename: "user"
  }
]
```
as previous result. And we can update the query to include the mutation result without any problems.

But if we use `options.reducer` like this: 
```
graphql(MY_QUERY, {
  options: () => ({
    reducer: (previousResult, action) => {
      if (action.type === 'APOLLO_MUTATION_RESULT' && action.operationName === 'MyMutation') {
...
```
the `previousResult` data is missing the `__typename` field.
Manually adding this field to the query works, but it should be automatically added.

**Comment by helfer** [comment]
@Torsten85 ok, I think that shouldn't be too hard to fix. Do you want to give it a try and make a PR!?

**Comment by helfer** [comment]
Fixed in #880
